packages/store-ui/src/atoms/Slider/Slider.tsx

```diff
@@ -0,0 +1,150 @@
+   */
+  testId?: string
+  /**
+   * Callback that fires when the slider value changes.
+   */
+  onChange?: (value: { min: number; max: number }) => void
+  /**
+   * Set a human-readable name for the slider.
+   */
+  ariaLabel?: string
+  /**
+   * A function used to set a human-readable value text based on the slider's current value.
+   */
+  getAriaValueText?(value: number, thumb?: 'min' | 'max'): string
+}
+
+const Slider = ({
+  min,
+  max,
+  onChange,
+  testId = 'store-slider',
+  showValues = true,
+  ariaLabel,
+  getAriaValueText,
+}: SliderProps) => {
+  const [minVal, setMinVal] = useState(min)
+  const [maxVal, setMaxVal] = useState(max)
+
+  const minValRef = useRef(min)
+  const maxValRef = useRef(max)
+  const range = useRef<HTMLDivElement>(null)
+
+  const getPercent = useCallback(
+    (value) => Math.round(((value - min) / (max - min)) * 100),
+    [min, max]
+  )
+
+  // width of the range to reduce from the left side
+  useEffect(() => {
+    const minPercent = getPercent(minVal)
+    const maxPercent = getPercent(maxValRef.current)
+
+    if (range.current) {
+      range.current.style.left = `${minPercent}%`
+      range.current.style.width = `${maxPercent - minPercent}%`
+    }
+  }, [minVal, getPercent])
+
+  // width of the range to reduce from the right side
+  useEffect(() => {
+    const minPercent = getPercent(minValRef.current)
+    const maxPercent = getPercent(maxVal)
+
+    if (range.current) {
+      range.current.style.width = `${maxPercent - minPercent}%`
+    }
+  }, [maxVal, getPercent])
+
+  useEffect(() => {
+    onChange?.({ min: minVal, max: maxVal })
+  }, [minVal, maxVal, onChange])
+
+  return (
+    <div
+      data-store-slider
+      data-testid={testId}
+      aria-label={ariaLabel}
+      // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
+      role="slider"
+    >
+      <div data-store-slider-range>
+        <div ref={range} data-store-slider-track />
+        <input
+          type="range"
+          min={min}
+          max={max}
+          value={minVal}
+          onChange={(event) => {
+            const value = Math.min(Number(event.target.value), maxVal - 1)
+
+            setMinVal(value)
+            minValRef.current = value
+          }}
+          data-store-slider-thumb="left"
+          style={{ zIndex: minVal > max - 100 ? 5 : 'auto' }}
+          aria-valuemin={min}
+          aria-valuemax={max}
+          aria-valuenow={minVal}
+          aria-label={String(minVal)}
+          aria-labelledby={
+            getAriaValueText ? getAriaValueText(minVal, 'min') : undefined
+          }
+        />
+        <input
+          type="range"
+          min={min}
+          max={max}
+          value={maxVal}
+          onChange={(event) => {
+            const value = Math.max(Number(event.target.value), minVal + 1)
+
+            setMaxVal(value)
+            maxValRef.current = value
+          }}
+          data-store-slider-thumb="right"
```

Comment:
Can you take a look at [this comment](https://github.com/vtex/faststore/pull/918/files#r697767728) I left on this other PR? I think it applies here as well.

Reply:
I don't understand 😕

Reply:
I think there should be `data-store-slider-thumb` for both thumbs, a `data-left` for the left one and `data-right` for the right one. As I've said on the linked comment, there's a while I don't review store UI components, so I may be wrong, but I think this deviates from the pattern we agreed upon.

Reply:
it's just the opposite, we should use absolute data-attribute only for when it's a component state control, here it's not involving state, just a variation, so we can use those variations as a value for `data-store-slider-thumb`.

Example:

If the thumb slider had the possibility of being disabled then it could have a `data-disabled`, but since there is no state control, then we don't need that kind of data-attribute.

You can read more about this in the Reach UI style documentation.

> There are certain states an element can go into like "selected". To target these styles use a "pseudo" selector (more like a pseudo-pseudo selector!). For example, to style all dropdown menu items in their highlighted, or selected state, use this CSS:

Reply:
Cool concept! I see what you mean. Thanks for detailing it :)